synapse/rest/media/v1/media_storage.py

```diff
@@ -145,15 +145,15 @@ async def finish() -> None:
                     f.flush()
                     f.close()
 
-                    spam = await self.spam_checker.check_media_file_for_spam(
+                    spam_check = await self.spam_checker.check_media_file_for_spam(
                         ReadableFileWrapper(self.clock, fname), file_info
                     )
-                    if spam:
+                    if isinstance(spam_check, Codes):
```

Comment:
In general I'd prefer if we test for whether `spam_check` equals `self.spam_checker.NOT_SPAM` to err on the safe side (e.g. if we decide to add a non-`Codes` value for denying events or operations later on we might forget to update some of these checks).

(this apply to other callbacks, not just `check_media_file_for_spam`)